packages/bvaughn-architecture-demo/components/sources/AutoComplete/AutoComplete.tsx

```diff
@@ -33,14 +34,9 @@ export default function AutoComplete({
       onChangeProp(newValue);
     }
 
-    const cursorIndex = input.selectionStart;
-    const shouldAutoComplete =
-      cursorIndex === newValue.length &&
-      newValue.length > 0 &&
-      newValue.charAt(newValue.length - 1) !== " ";
-    const expression = shouldAutoComplete
-      ? getExpressionFromString(newValue, cursorIndex - 1)
-      : null;
+    const cursorIndex = input.selectionStart || value.length;
+    const shouldAutoComplete = newValue.length > 0 && newValue.charAt(newValue.length - 1) !== " ";
+    const expression = shouldAutoComplete ? getExpressionFromString(newValue, cursorIndex) : null;
```

Comment:
:question: this means we oughta be able to replace use of Babel for parsing in the JSTerm as well, right?

Reply:
Yes! 😄 Once that project is officially under way.

Reply:
TBH this feels so close now...all that's missing is syntax highlighting in edit mode.